### ISSUE TITLE
Sentry security concern

### ISSUE THREAD

**Issue by epicfaace** [codalab/codalab-worksheets#3831]
Punting on this issue for now. Should look into security concern wherein any user could flood the Sentry logs because they could access the CODALAB_SENTRY_INGEST_URL which is public, so we should determine whether we want to expose the Sentry to the server as well as the client.

_Originally posted by @andyjin2000 in https://github.com/codalab/codalab-worksheets/issues/3536#issuecomment-942946148_

**Comment by epicfaace** [comment]
@percyliang @andyjin2000 @pranavjain See this thread for a response from Sentry about this concern: https://forum.sentry.io/t/how-does-sentry-prevent-spammers/8188/2

**Comment by epicfaace** [comment]
Also see https://github.com/getsentry/sentry/issues/15877

**Comment by epicfaace** [comment]
@pranavjain did you get a chance to look into these responses? What do you think?

**Comment by pranavjain** [comment]
I did go through the links.

I am still not convinced by the statement:
> That’s pretty much the state of the art and there is no way to automatically prevent it. We haven’t heard of a practical issue with this. 

Why can't someone flood our logs? What is preventing them? I understand we can do IP-based filtering etc., but that won't stop them from flooding it in the first place.

Can we send the logs to the backend and post them to Sentry from there? [Sentry documentation](https://docs.sentry.io/product/sentry-basics/dsn-explainer/#dsn-utilization) has details about allocating DSN dynamically and we can explore that option too.

**Comment by epicfaace** [comment]
There's a tradeoff between complete theoretical security and usability / efficiency. I think your suggestion (send the logs to the backend and post them to Sentry from there) would go more in the complete theoretical security direction, but would cause problems with usability / efficiency -- we don't want every single event to go through our server, as that would require us to have a larger load. And if we did that, someone could just flood our server and take it down.

I think it's important to think about what's the worst-case scenario here: someone floods our logs. This may be a headache for Sentry (and they likely already have processes to rate-limit it), so I don't see why this is really a problem we should correct for.

Also, this concern you raise also applies to other common third-party analytics solutions, for example, Google Analytics: https://stackoverflow.com/questions/15468601/google-analytics-javascript-security-property-id-public. It's standard practice that this is an OK tradeoff.

**Comment by adiprerepa** [comment]
I grappled with this issue back when I implemented Sentry with CodaLab, and I'm not sure there is anything we can do. It's definitely not ideal, but the alternatives are too cumbersome/may oversolve an issue that may not exist in the first place. Besides, I think our logs being flooded in the first place not that big of a deal...we would just temporarily revert to the way we observed our system pre-sentry.
